### PR TITLE
log/correct preference mismatches

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/services-team 
https://wikia-inc.atlassian.net/browse/SERVICES-660
https://wikia-inc.atlassian.net/browse/SERVICES-736
https://wikia-inc.atlassian.net/browse/SERVICES-737

when calling `loadOptions` in the `User` class, compare `User->mOptions` to the output of `PreferenceService`, log any differences, and then correct the data in the `PreferenceService`. Also piggybacking in tests for `PreferenceScopeService`

### CODE REVIEW

File: lib/Wikia/src/Service/User/Preferences/Migration/PreferenceCorrectionService.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		$differences = 0;
+
+		foreach ( $options as $name => $value ) {
+			if ( $this->scopeService->isGlobalPreference( $name ) ) {
+				$expectedPreferences->setGlobalPreference( $name, $value );
+
+				if ( !$actualPreferences->hasGlobalPreference( $name ) ||
+					$actualPreferences->getGlobalPreference( $name ) !== $value ) {
+
+					$this->logMissingPreference( $userId, $name, $value, $actualPreferences->getGlobalPreference( $name ) );
```
Comment (ArturKlajnerok) [inline lib/Wikia/src/Service/User/Preferences/Migration/PreferenceCorrectionService.php]: This seems to log mismatch value or missing preference, should we split it to have more precise logs ?

**Comment by ArturKlajnerok** [comment]
Other then one small clarification, LGTM :+1:

File: lib/Wikia/src/Service/User/Preferences/Migration/PreferenceCorrectionService.php (hunk 1 above)
Comment (nmonterroso) [inline lib/Wikia/src/Service/User/Preferences/Migration/PreferenceCorrectionService.php]: That's a good idea. I'll do that.

Comment (nmonterroso) [inline lib/Wikia/src/Service/User/Preferences/Migration/PreferenceCorrectionService.php]: d681ea1

**Pull request merged**